src/libraries/PositionInfoLibrary.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/src/types/PoolId.sol";
+
+/**
+ * @dev PositionInfo is a packed version of solidity structure.
+ * Using the packaged version saves gas and memory by not storing the structure fields in memory slots.
+ *
+ * Layout:
+ * 200 bits poolId | 24 bits tickUpper | 24 bits tickLower | 8 bits hasSubscriber
+ *
+ * Fields in the direction from the least significant bit:
+ *
+ * A flag to know if the tokenId is subscribed to an address
+ * uint8 hasSubscriber;
+ *
+ * The tickUpper of the position
+ * int24 tickUpper;
+ *
+ * The tickLower of the position
+ * int24 tickLower;
+ *
+ * The truncated poolId. Truncates a bytes32 value so the most signifcant (highest) 200 bits are used.
+ * bytes25 poolId;
+ *
+ * Note: If more bits are needed, hasSubscriber can be a single bit.
+ *
+ */
+type PositionInfo is uint256;
+
+library PositionInfoLibrary {
+    using PoolIdLibrary for PoolKey;
+
+    PositionInfo internal constant EMPTY_POSITION_INFO = PositionInfo.wrap(0);
+
+    uint256 internal constant MASK_UPPER_200_BITS = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF00000000000000;
+    uint256 internal constant MASK_8_BITS = 0xFF;
+    uint24 internal constant MASK_24_BITS = 0xFFFFFF;
+    uint256 internal constant SET_UNSUBSCRIBE = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF00;
+    uint256 internal constant SET_SUBSCRIBE = 0x01;
+    uint8 internal constant TICK_LOWER_OFFSET = 8;
+    uint8 internal constant TICK_UPPER_OFFSET = 32;
+
+    function poolId(PositionInfo info) internal pure returns (bytes25 _poolId) {
+        assembly ("memory-safe") {
+            _poolId := and(MASK_UPPER_200_BITS, info)
+        }
+    }
```

Comment:
Lets add a natspec saying `info.poolId()` is NOT compatible with `PoolId` on core.

i'm tempted to suggest calling it something else like `truncatedPoolId()` but 🤮

Reply:
Yeah good call out, I'll add natspec 😛